### PR TITLE
Fixed #33543 -- Depracated passing False to OrderBy's nulls_first and nulls_last.

### PR DESCRIPTION
**Pull request by AllenJonathan**

### CODE REVIEW

File: django/db/models/expressions.py [hunk 1]
```diff
@@ -1489,11 +1489,11 @@ class OrderBy(Expression):
...
     conditional = False
 
-    def __init__(
-        self, expression, descending=False, nulls_first=False, nulls_last=False
-    ):
+    def __init__(self, expression, descending=False, nulls_first=None, nulls_last=None):
         if nulls_first and nulls_last:
             raise ValueError("nulls_first and nulls_last are mutually exclusive")
+        if nulls_first is False or nulls_last is False:
+            raise ValueError("nulls_first and nulls_last cannot be False")
```
Comment (felixxm) [inline django/db/models/expressions.py:1496]: The current support for passing `False` to the `nulls_first` and `nulls_last` should be first deprecated and not removed immediately, see [docs](https://docs.djangoproject.com/en/4.0/internals/contributing/writing-code/submitting-patches/#deprecating-a-feature-1).

Comment (AllenJonathan) [reply django/db/models/expressions.py:1496]: So based on my understanding - 

- Only a warning should be given for passing False in null_first and nulls_last (any idea on what the warning could be would be appreciated).
- The improvements should not be done for the time being (should be done later).

Am I right?

Comment (felixxm) [reply django/db/models/expressions.py:1496]: > Only a warning should be given for passing False in null_first and nulls_last (any idea on what the warning could be would be appreciated).

For example `Passing False to nulls_first/nulls_last arguments is deprecated, use None instead.`

> The improvements should not be done for the time being (should be done later).

Yes, see e.g. 31174031f1ded30d96c77908b965755e0be94c94.

**Comment by felixxm** [comment]
@AllenJonathan Do you have time to keep working on this?

**Comment by AllenJonathan** [comment]
I'm busy for a week, (university exams going on). Perhaps can do it after that. If this needs to be done quickly or anyone wants to take over they can.

**Comment by felixxm** [comment]
Superseded by #15682.

**Pull request closed without merging**